### PR TITLE
[EOS-S3] Sumbodules

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR adds Git submodules needed for QuickLogic EOS-S3 eFPGA support

### CODE REVIEW

**Comment by mkurc-ant** [comment]
Strange, This PR just adds a bunch of new submodules and the CI fails on some seemingly unrelared `xc7` tests. There are multiple failures with the same DRC violation during Vivado bitgen:

```
FAILED: cd /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/build/xc/xc7/tests/ddr/ddr_uart_arty100t/artix7_100t-xc7a100t-virt-xc7a100t-test/interchange && /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/env/conda/envs/symbiflow_arch_def_base/bin/cmake -E env XRAY_VIVADO_SETTINGS=/image/Xilinx/Vivado/2019.2/settings64.sh /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/third_party/prjxray/utils/vivado.sh -mode batch -source /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/build/xc/xc7/tests/ddr/ddr_uart_arty100t/artix7_100t-xc7a100t-virt-xc7a100t-test/interchange/ddr_uart_arty100t_vivado_interchange_runme.tcl > /tmpfs/src/github/symbiflow-arch-defs-presubmit-xc7-vendor/build/xc/xc7/tests/ddr/ddr_uart_arty100t/artix7_100t-xc7a100t-virt-xc7a100t-test/interchange/vivado.stdout.log
ERROR: [DRC PLCR-1] Placement Constraints Check for Clock Region(s): Design Check found an error in Clock Region X1Y1. This clock region has 13 clocks locked whereas only 12 clocks can be routed per clock region. The cells were either constrained through Area Groups or locked via user constraints. This situation can be resolved by unlocking some cells from the user constraints so that there are only 12 clocks in this region. 
ERROR: [Vivado 12-1345] Error(s) found during DRC. Bitgen not run.
```

File: .gitmodules [hunk 1]
```diff
@@ -70,3 +70,18 @@
...
 	url = https://github.com/QuickLogic-Corp/ql_designs.git
+[submodule "third_party/ql-eos-s3"]
+	path = third_party/ql-eos-s3
+	url = https://github.com/QuickLogic-Corp/EOS-S3
+[submodule "third_party/ql-pp3e"]
+	path = third_party/ql-pp3e
+	url = https://github.com/QuickLogic-Corp/PolarPro3E.git
+[submodule "quicklogic/common/utils/fasm-utils"]
+	path = quicklogic/common/utils/fasm-utils
+	url = https://github.com/QuickLogic-Corp/quicklogic-fasm-utils.git
```
Comment (acomodi) [inline .gitmodules:81]: I wonder if this can be actually generalized and upstreamed into the fasm library?

**Review by acomodi — APPROVED** [review]
LGTM

File: .gitmodules (hunk 1 above)
Comment (mkurc-ant) [reply .gitmodules:81]: Indeed it could. This one is used only by `quicklogic-fasm` so probably we can integrate both into one.

**Pull request merged**